### DISCUSSION TITLE
Request for access to Vercel WAF Rate Limit Beta

### DISCUSSION THREAD

**Discussion by Sakura-sx** [vercel/community#7163] [Feature Question]
### Question

Hello, is there any way I could get access to Vercel WAF Rate limit beta? I have been getting a lot of attacks and even if I am there and I can enable rules or challenge mode, the billing goes insane. If there is no way, could I set rules or challenge mode with an API? I really like the job that is being done in the firewall. I is amazing. Thanks in advance!

**Comment by leerob** [comment]
Hey there, can you share more about how you're currently blocking attacks? 

If you are challenging all traffic, wouldn't that be preventing malicious traffic from coming through?

Rate limiting is in a private beta right now, but will be opening up further in the future!

[a comment by an automated account is not shown]
